spec/game_runner_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+require_relative '../game_runner.rb'
+
+RSpec.describe GameRunner do
+  def grid_template
+    <<~GRID
+   __ __ __
+  |  |  |  |
+  |__|__|__|
+  |  |  |  |
+  |__|__|__|
+  |  |  |  |
+  |__|__|__|
+    GRID
+  end
+
+  describe '#run' do
+    it 'on start up, an empty grid is rendered on the screen' do
+      stdout = StringIO.new(' ')
+      @game_start = GameRunner.new(stdout)
+
+      game = @game_start.run
```

Comment:
Sorry I didn't notice this the first time through. The run action shouldn't return anything.

This is a method that does something, not return something. That distinction is called the command/query separation. [https://martinfowler.com/bliki/CommandQuerySeparation.html](https://martinfowler.com/bliki/CommandQuerySeparation.html)

The whole reason we pass something in is so we have a handle on an object that understands what happened in there. So you should be working with the object you passed into the game, not the return value.

Reply:
Ohhhh!!! Okay, got it! I see that command methods are also usually named as verbs. The `run` command is modifying `stdout`, so it's `stdout` that I want to make sure has changed!

Reply:
YES!